lib/plugins/pluginUtils.js

```diff
@@ -106,8 +106,14 @@ function createProcessAttribute(fn, attributeType) {
         }
 
         if (config.checkNgsi2()) {
-            if (attribute.metadata) {
+
+            if (Array.isArray (attribute.metadata)){
                 attribute.metadata = attribute.metadata.map(createProcessAttribute(fn, attributeType));
+            } else if (attribute.metadata) {
+                Object.keys(attribute.metadata).forEach(function(key) {
+                    var entry = attribute.metadata[key];
+                    entry.value = fn(entry.value);
+                });
```

Comment:
I guess this modification is related with:

> For some reason the code made the assumption that metadata within an NGSI v2 attribute was an **array**, which I believe is incorrect (or a hangover from NGSI v1 `metadatas`) - I've added an Array check in `pluginUtils.js` to maintain the backwards functionality whilst also being able to process metadata Objects properly.

I think is the later (a hanover from NGSIv1 metadatas).

Given you have prepared the code to cover both cases (great!), maybe it a good idea to include a commenta about it (something like: "This code is prepared to process metadata in NGSIv1-style (array) and NGSIv2-style (object)"). This way it would be easy to remove the NGSIv1 part if we remove at the end all NGSIv1 stuff from agents code.

Reply:
Clarification added. Fixed a2d139